compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Env.hs

```diff
@@ -56,14 +58,14 @@ getWorld = view world
 
 -- | Type class constraint capturing the needed monadic effects for the
 -- functions manipulating the type checker environment.
-type MonadGamma m = (MonadError Error m, MonadReader Gamma m)
+type MonadGamma m = (MonadError Error m, MonadReader Gamma m, MonadState [Warning] m)
```

Comment:
ah, I'm guessing you went with `MonadState` instead of `MonadWriter` to avoid the leaky `WriterT` implementations, right? I see our version of `transformers` has `Control.Monad.Trans.Writer.CPS` but our `mtl` doesn't have `MonadWriter` instances for it :( oh well, we can take care of this later

Reply:
Yep - though I'm not sure how much that matters given that a warning list shouldn't ever be particularly large. Also, considering that some warnings may need to be upgraded to errors (this is the model that GHC follows that I'd like us to follow eventually) I thought it best to leave things introspectable at runtime.